ui/src/bootstrap.rs

```diff
@@ -59,18 +62,24 @@ pub(crate) fn use_boostrap<'a>(
 
 pub fn set_app_panic_hook() {
     panic::set_hook(Box::new(|panic_info| {
-        let intro = match panic_info.payload().downcast_ref::<&str>() {
-            Some(s) => format!("panic occurred: {s:?}"),
-            None => "panic occurred".into(),
-        };
-        let location = match panic_info.location() {
-            Some(loc) => format!(" at file {}, line {}", loc.file(), loc.line()),
-            None => "".into(),
-        };
-
         let logs = logger::dump_logs();
-        let crash_report = format!("{intro}{location}\n{logs}\n");
+        let crash_report = format!("{panic_info}\n{logs}\n");
+        let backtrace = Backtrace::force_capture();
+        let time = Local::now().format("%d-%m-%Y_%H-%M-%S").to_string();
+        let path = STATIC_ARGS.crash_logs.join(format!("{}.log", time));
+        let log = format!(
+            "Uplink crashed:\n{}\nBackTrace:\n{}",
+            crash_report, backtrace
+        );
         println!("{crash_report}");
+        if let Some(p) = path.parent() {
+            if fs::create_dir_all(p).is_ok() {
+                match fs::write(path.clone(), log) {
```

Comment:
If I recall, `std::fs::write` accepts `impl AsRef<Path>`. If thats the case, we can do `&path` or `path` but since you are also reusing it, the reference should suffice.